enumerables.rb

```
@@ -0,0 +1,156 @@
+      my_each { |n| return true if n.is_a?(arg) == true }
+    elsif !arg.nil? && arg.class == Regexp
+      my_each { |n| return true if arg.match(n) }
+    else
+      my_each { |n| return true if n == arg }
+    end
+    false
+  end
+
+  def my_none?(arg = nil)
+    if !block_given? && arg.nil?
+      my_each { |n| return false if n }
+      return true
+    end
+
+    if !block_given? && !arg.nil?
+
+      if arg.is_a?(Class)
+        my_each { |n| return false if n.class == arg }
+        return true
+      end
+
+      if arg.class == Regexp
+        my_each { |n| return false if arg.match(n) }
+        return true
+      end
+
+      my_each { |n| return false if n == arg }
+      return true
+    end
+
+    my_any? { |item| return false if yield(item) }
+    true
+  end
+
+  def my_count(arg = nil)
+    count = 0
+    if block_given?
+      to_a.my_each { |i| count += 1 if yield(i) }
+    elsif !arg.nil?
+      to_a.my_each { |i| count += 1 if i == arg }
+    else
+      to_a.my_each { |i| count += 1 if i }
+    end
+    count
+  end
+
+  def my_map(proc = nil)
+    return to_enum(:my_map) unless block_given?
+
+    result = []
+    my_each { |element| result << proc.call(element) } if block_given? && proc
+    my_each { |element| result << yield(element) } if proc.nil?
+    result
+  end
+
+  def my_inject(number = nil, symbol = nil)
+    if block_given?
+      accum = number
+      my_each do |item|
+        accum = accum.nil? ? item : yield(accum, item)
+      end
+      accum
+    elsif !number.nil? && (number.is_a?(Symbol) || number.is_a?(String))
+      accum = nil
+      my_each do |item|
+        accum = accum.nil? ? item : accum.send(number, item)
+      end
+      accum
+    elsif !symbol.nil? && (symbol.is_a?(Symbol) || symbol.is_a?(String))
+      accum = number
+      my_each do |item|
+        accum = accum.nil? ? item : accum.send(symbol, item)
+      end
+      accum
+    else
+      LocalJumpError.new 'no block given'
```

`raise LocalJumpError, 'no block is given', caller` instead of returning an obect.